### PR TITLE
Handle logging story better on daemon

### PR DESCRIPTION
**Pull request by arlyon**
### Description

This changes the config from hard-coded to take the setup from your env vars, defaulting to info if unset. It also prevents the daemon from logging span lines. It also adds the ability to clean logs when running `daemon clean` so that users can more easily manage their logs.

### Testing Instructions

Run `turbo daemon` with `TURBO_LOG_VERBOSITY` verbosity unset or set to trace to observe the difference. Run `turbo clean` and see the logs have been removed.

Closes TURBO-2450

### CODE REVIEW

File: crates/turborepo-lib/src/commands/daemon.rs [hunk 1]
```diff
@@ -133,7 +133,7 @@ pub async fn daemon_client(command: &DaemonCommand, base: &CommandBase) -> Resul
                 .status()
                 .expect("failed to execute tail");
         }
-        DaemonCommand::Clean => {
+        DaemonCommand::Clean { clean_logs: logs } => {
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/daemon.rs:136]:
```suggestion
        DaemonCommand::Clean { clean_logs: should_clean_logs } => {
```

File: crates/turborepo-lib/src/commands/daemon.rs [hunk 2]
```diff
@@ -193,6 +196,26 @@ async fn clean(
...
+        println!("Please remove manually: {}", log_folder);
+        success = false;
+    };
+
+    if success {
+        Ok(())
+    } else {
+        // return error
+        Err(DaemonError::CleanFailed)
+    }
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/daemon.rs:216]:
```suggestion
    trace!("cleaning up log files");
    // clear all files in the log folder. we want to keep the
    // folder just remove the contents
    // `remove_dir_all_recursive` is lifted from `std`
    fs::remove_dir_all(log_folder.as_std_path()).map_err(|e|  {
        println!("Failed to remove log files: {}", e);
        println!("Please remove manually: {}", log_folder);
        DaemonError::CleanFailed
    })
```

File: crates/turborepo-lib/src/commands/daemon.rs [hunk 3]
```diff
@@ -193,6 +196,26 @@ async fn clean(
     }
 }
 
+async fn clean_logs(log_folder: &AbsoluteSystemPath) -> Result<(), DaemonError> {
+    let mut success = true;
+    trace!("cleaning up log files");
+    // clear all files in the log folder. we want to keep the
+    // folder just remove the contents
+    // `remove_dir_all_recursive` is lifted from `std`
+    if let Err(e) = fs::remove_dir_all(log_folder.as_std_path()) {
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/daemon.rs:205]: I think we should generally prefer using the FS ops through our path lib as we use `fs_err` which includes the path name in the error.
```suggestion
    if let Err(e) = log_folder.remove_dir_all() {
```

File: crates/turborepo-lib/src/commands/daemon.rs [hunk 4]
```diff
@@ -193,6 +196,26 @@ async fn clean(
     }
 }
 
+async fn clean_logs(log_folder: &AbsoluteSystemPath) -> Result<(), DaemonError> {
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/daemon.rs:199]: I don't think this needs to be async
```suggestion
fn clean_logs(log_folder: &AbsoluteSystemPath) -> Result<(), DaemonError> {
```

**Review by chris-olszewski — APPROVED** [review]
Cleaning the daemon logs seems like a good change, but not what the PR title & body suggest. Totally think this should get merged in, but we should probably update the title so our changelog is more accurate.

Also some unused imports that should get removed before merge.

**Review by mehulkar — APPROVED** [review]

**Pull request merged**